katdal/h5datav2_5.py

```diff
@@ -0,0 +1,631 @@
+        # Move timestamps from start of each dump to the middle of the dump
+        data_timestamps += 0.5 * self.dump_period + self.time_offset
+
+        if data_timestamps[0] < 1e9:
+            logger.warning("File '%s' has invalid first timestamp (%f)" % (filename, data_timestamps[0],))
+        self._time_keep = np.ones(num_dumps, dtype=np.bool)
+        self.start_time = katpoint.Timestamp(data_timestamps[0] - 0.5 * self.dump_period)
+        self.end_time = katpoint.Timestamp(data_timestamps[-1] + 0.5 * self.dump_period)
+        self._keepdims = keepdims
+
+        # ------ Extract flags ------
+        # Check if flag group is present, else use dummy flag data
+        self._flags = markup_group['flags'] if 'flags' in markup_group else \
+            dummy_dataset('dummy_flags', shape=self._vis.shape[:-1], dtype=np.uint8, value=0)
+        # Obtain flag descriptions from file or recreate default flag description table
+        self._flags_description = markup_group['flags_description'] if 'flags_description' in markup_group else \
+            np.array(zip(FLAG_NAMES, FLAG_DESCRIPTIONS))
+
+        # ------ Extract sensors ------
+        # Populate sensor cache with all HDF5 datasets below sensor group that fit the description of a sensor
+        cache = {}
+
+        def register_sensor(name, obj):
+            """A sensor is defined as a non-empty dataset with expected dtype."""
+            if isinstance(obj, h5py.Dataset) and obj.shape != () and obj.dtype.names == ('timestamp', 'value', 'status'):
+                cache[name] = RecordSensorData(obj, name)
+        sensors_group.visititems(register_sensor)
+        # Use estimated data timestamps for now, to speed up data segmentation
+        self.sensor = SensorCache(cache, data_timestamps, self.dump_period, keep=self._time_keep,
+                                  props=SENSOR_PROPS, virtual=VIRTUAL_SENSORS, aliases=SENSOR_ALIASES)
+
+        # ------ Extract subarrays ------
+        # By default, only pick antennas that were in use by the script
+        script_ants = str(config_group['Observation'].attrs['ants']).split(',')
+
+        self.ref_ant = script_ants[0] if not ref_ant else ref_ant
+        # Original list of correlation products as pairs of input labels
+        corrprods = get_single_value(config_group["DBE"], "vis_ordering").split(',')
+        if len(corrprods) != self._vis.shape[2]:
+            raise BrokenFile('Number of data labels (containing expected antenna names) '
+                             'received from h5 file (%d) differs from number of power products in data (%d)' %
+                             (len(corrprods), self._vis.shape[2]))
+        # Get the corrprod labels into the format that KatDAL wants, the v2.5 files only give ll and rr.
+        # This will change it into  "ant1lant1l,ant1rant1r" which katdal expects.
+        corrprods = [('ant1' + corrprods[0][0], 'ant1' + corrprods[0][1]),
```

Comment:
Is it always 'ant1'? Maybe better to do a constant at the top for this.

Reply:
Noted.

Reply:
Also, what if the antenna name in the file attributes differs from 'ant1' (not likely but plausible)?

Reply:
Reworked in b6ff28a.